python/cudf/cudf/core/frame.py

```diff
@@ -38,9 +40,23 @@ class Frame(libcudf.table.Table):
     """
 
     @classmethod
-    def _from_table(cls, table):
+    def _from_table(cls, table: "Frame"):
         return cls(table._data, index=table._index)
 
+    @overload
+    def _mimic_inplace(self, result: "Frame") -> "Frame":
+        ...
+
+    @overload
+    def _mimic_inplace(self, result: "Frame", inplace: Literal[True]):
+        ...
+
+    @overload
+    def _mimic_inplace(
+        self, result: "Frame", inplace: Literal[False]
+    ) -> "Frame":
+        ...
```

Comment:
Can we not use something like `Optiona[Literal[False]]` to avoid having to have an additional overload?

Reply:
It looks like `Optional[Literal[False]]` is the same as `Literal[False, None]` (i.e., literally `False` or literally `None`). So this doesn't work unfortunately :(